File: src/main/java/com/NovumScientiaTeam/novumscientia/api/materialProperties/transfer/FluidTransferProperties.java
package com.NovumScientiaTeam.novumscientia.api.materialProperties.transfer;

import com.EmosewaPixel.pixellib.materialSystem.materials.Material;

import java.util.HashMap;
import java.util.Map;

public class FluidTransferProperties extends TransferProperties {
    private static final Map<Material, FluidTransferProperties> itemTransferMap = new HashMap<>();

    public static void setMaterialProperties(Material mat, int fluidCapacity, int mbsPerTick) {
        itemTransferMap.put(mat, new FluidTransferProperties(fluidCapacity, mbsPerTick));
    }

    public static FluidTransferProperties getPropertiesForMaterial(Material mat) {
        return itemTransferMap.get(mat);
    }

    private FluidTransferProperties(int fluidCapacity, int mbsPerTick) {
        super(fluidCapacity, mbsPerTick);
    }
}

File: src/main/java/com/NovumScientiaTeam/novumscientia/api/materialProperties/transfer/FluidTransferProperties.kt
package com.novumscientiateam.novumscientia.api.materialproperties.transfer

import com.emosewapixel.pixellib.materialsystem.materials.Material

import java.util.HashMap

class FluidTransferProperties private constructor(fluidCapacity: Int, mbsPerTick: Int) : TransferProperties(fluidCapacity, mbsPerTick) {
    companion object {
        private val itemTransferMap = HashMap<Material, FluidTransferProperties>()

        fun setMaterialProperties(mat: Material, fluidCapacity: Int, mbsPerTick: Int) {
            itemTransferMap[mat] = FluidTransferProperties(fluidCapacity, mbsPerTick)
        }

        fun getPropertiesForMaterial(mat: Material) = itemTransferMap[mat]
    }
}